.github/workflows/examples.yml

```diff
@@ -0,0 +1,13 @@
+name: examples
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+  workflow_dispatch:
+
```

Comment:
Would be nice for one of these workflows to allow the ability to configure a timeout https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

This would protect us from a change in examples (or the CLI) that accidentally leaves something hanging, and then racks up our workflow minutes (the default is 6 hours). There was actually an Artillery CLI bug that has resurfaced 2 times already IIRC that would leave the CLI hanging indefinitely, for example.

The only thing is if we want to make it configurable per test (would be nice as some tests may require longer runs - these are load tests after all), or just do a big enough timeout that should work for all cases.

Reply:
Yeah, this makes perfect sense. I wonder if GitHub will respect a workflow-based timeout setting when reusing workflows. Can test that out.

Reply:
Since the purpose of automated tests is to be fast, I don't think any test should take a significant time to run. A long-running test means long CIs and slow iteration speed. We likely don't need any long-running tests in these workflows to begin with. 

I understand some load test scenarios will correlate with long running time, but I'd go with the positive expectation here that forces CI to be fast and then adjust when and if we arrive at such tests. 

I will set the `timeout-minutes` to 60, accounting for unterminated test runs not bumping our usage metrics.

Reply:
I set timeout to 10 minutes. Happy to iterate on this if our examples ever exceed that (they really shouldn't).